hilti/src/rt/tests/regexp.cc

```diff
@@ -83,6 +83,25 @@ TEST_SUITE_END();
 
 TEST_SUITE_BEGIN("MatchState");
 
+TEST_CASE("advance") {
+    CHECK_EQ(RegExp("123", regexp::Flags()).tokenMatcher().advance("123"_b, false), std::make_tuple(-1, 3));
+    CHECK_EQ(RegExp("123", regexp::Flags()).tokenMatcher().advance("123"_b, true), std::make_tuple(1, 3));
+
+    CHECK_EQ(RegExp(std::vector<std::string>({"abc", "123"}), regexp::Flags()).tokenMatcher().advance("123"_b, true),
+             std::make_tuple(2, 3));
+
+    CHECK_EQ(RegExp("", regexp::Flags()).tokenMatcher().advance("123"_b, false), std::make_tuple(-1, 3));
```

Comment:
an empty regex should always match (or be rejected, either way would be fine)

Reply:
Is returning `(-1, 3)` for 3 bytes of input not the correct answer for _always match_? The range seems also valid, but if we match always, we also always can consume more data.

Reply:
I would expect an empty regex  to immediately match an empty string and stop there, so `(1, 0)`. 

Not a big deal, but worth another TODO.